### ISSUE TITLE
ListApplicationUsers not working

### ISSUE THREAD

**Issue by koswedo** [okta/okta-sdk-dotnet#602]
### Describe the bug?

when calling ListApplicationUsers from ApiClient it fails with error
> The operation has timed out.

While debugging, I noticed that the path is set to `/api/v1/apps/{appId}/users`. `{appId}` does not seem to be replaced by the value.
If I copy the ApiClient into my project to replace `{appId}` by the actual value, the request is passing through.

### What is expected to happen?

To get a list of application users

### What is the actual behavior?

a `The operation has timed out` error

### Reproduction Steps?

call ListApplicationUsers from ApiClient

### .NET Version

6.0.402

### SDK Version

6.0.0

### OS version

BuildNumber Caption OSArchitecture Version
19042 Microsoft Windows 10 Enterprise 64-bit 10.0.19042

**Comment by bryanapellanes-okta** [comment]
@koswedo Thank you for reporting this issue. I filed an internal ticket to be reviewed and prioritized by the team.

Internal Ref: OKTA-549675

**Comment by laura-rodriguez** [comment]
Hi @koswedo,

Can you please provide more details about your users and apps? We do have [a test that covers this particular scenario](https://github.com/okta/okta-sdk-dotnet/blob/master/src/Okta.Sdk.IntegrationTest/ApplicationScenarios.cs#L905), and it's working fine. There might be some variation in your use case that we need to know to reproduce this issue. 

I wonder if this is related to #599? Thank you!

**Comment by koswedo** [comment]
@laura-rodriguez
Thank you for your answer. 

To confirm that the issue is coming from your code and not mine (well, it can happen from time to time ;) ), I have copied your ApiClient.cs and changed lines [300](https://github.com/okta/okta-sdk-dotnet/blob/41ac3fde0b83b8eb0a708990cf843b119fa18dc5/src/Okta.Sdk/Client/ApiClient.cs#L300) to 306
from
```cs
  if (options.PathParameters != null)
  {
      foreach (var pathParam in options.PathParameters)
      {
          request.AddParameter(pathParam.Key, pathParam.Value, ParameterType.UrlSegment);
      }
  }
```
to 
```cs
if (options.PathParameters != null)
{
    foreach (var pathParam in options.PathParameters)
    {
        request.Resource = request.Resource.Replace($"{{{pathParam.Key}}}", pathParam.Value, StringComparison.Ordinal);
    }
}
```
and the error disappears and fall back to issue [#599](https://github.com/okta/okta-sdk-dotnet/issues/599). It seems the `{appId}` part of the path is not substituted.

Let's be clear: I am not suggesting you should fix the issue with the code I have created. This is just to show where the issue is located.

I hope this help

**Comment by koswedo** [comment]
Hi @laura-rodriguez,
I have finally found out what the issue is. The code is working fine but the documentation is misleading. 
In the [readme.md](https://github.com/okta/okta-sdk-dotnet/blob/41ac3fde0b83b8eb0a708990cf843b119fa18dc5/README.md?plain=1#L442) file and in [Configuration.cs](https://github.com/okta/okta-sdk-dotnet/blob/41ac3fde0b83b8eb0a708990cf843b119fa18dc5/src/Okta.Sdk/Client/Configuration.cs#L153), `ConnectionTimeout` units are seconds, so I set the value to 60 thinking it would 60s. I realized the units is not seconds but milliseconds as mentioned in [IReadableConfiguration.cs](https://github.com/okta/okta-sdk-dotnet/blob/41ac3fde0b83b8eb0a708990cf843b119fa18dc5/src/Okta.Sdk/Client/IReadableConfiguration.cs#L109). So this is why I got all the time, with good reason, a time-out error.
So why when I changed the ApiClient, it worked, you ask me? well, it is a mistake on my side, for any reason, the `ConnectionTimeout` wasn't set in this case.
So, the issue is not the code but with the documentation. How to you want address this? Should I change the title of the issue? Should I close this one and open another one? Do you take care of it?
Let me know

**Comment by koswedo** [comment]
A suggestion for `ConnectionTimeout`: using a `TimeSpan` would avoid confusion in the future. Just a thought.

**Comment by laura-rodriguez** [comment]
Hi @koswedo, 

Thanks for looking into it.  I'll take care of this.

**Comment by NoahStolk** [comment]
The documentation is indeed incorrect. I also spent quite a while figuring out why I was getting timeouts before finding this thread.

Is it ok to open a PR fixing these summaries?

https://github.com/okta/okta-sdk-dotnet/blob/41ac3fde0b83b8eb0a708990cf843b119fa18dc5/src/Okta.Sdk/Client/Configuration.cs#L148

https://github.com/okta/okta-sdk-dotnet/blob/41ac3fde0b83b8eb0a708990cf843b119fa18dc5/src/Okta.Sdk/Client/Configuration.cs#L156

**Comment by laura-rodriguez** [comment]
Hi @NoahStolk,

Apologize for the inconvenience. We have an open PR that fixes the timeout doc among other issues: #608

**Comment by NoahStolk** [comment]
Hi @laura-rodriguez, thanks for the reply. Good to see the migration guide being updated. 👍 Judging from the PR title, this only updates the documentation files and won't fix the C# documentation comments (summary and other elements) right?

I think it could be a good idea to include the documentation fixes in a future minor release, so the documentation displayed in an IDE won't be misleading.

**Comment by laura-rodriguez** [comment]
Fixed by #608
